plugin/plugins/filesystem/nodes/ipfs.go

```diff
@@ -0,0 +1,222 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/djdv/p9/p9"
+	files "github.com/ipfs/go-ipfs-files"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+// IPFS exposes the IPFS API over a p9.File interface
+// Walk does not expect a namespace, only its path argument
+// e.g. `ipfs.Walk([]string("Qm...", "subdir")` not `ipfs.Walk([]string("ipfs", "Qm...", "subdir")`
+type IPFS struct {
+	IPFSBase
+	ipfsHandle
+}
+
+func IPFSAttacher(ctx context.Context, core coreiface.CoreAPI) *IPFS {
+	id := &IPFS{IPFSBase: newIPFSBase(ctx, rootPath("/ipfs"), p9.TypeDir,
+		core, logging.Logger("IPFS"))}
+	id.meta, id.metaMask = defaultRootAttr()
+	return id
+}
+
+func (id *IPFS) Attach() (p9.File, error) {
+	id.Logger.Debugf("Attach")
+	//TODO: check core connection here
+	return id, nil
+}
+
+func (id *IPFS) GetAttr(req p9.AttrMask) (p9.QID, p9.AttrMask, p9.Attr, error) {
+	id.Logger.Debugf("GetAttr")
+	id.Logger.Debugf("GetAttr path: %v", id.Path)
+
+	if id.Path.Namespace() == nRoot { // metadata should have been initialized by attacher, don't consult CoreAPI
+		return id.Qid, id.metaMask, id.meta, nil
+	}
+
+	if err := coreGetAttr(id.Ctx, &id.meta, req, id.core, id.Path); err != nil {
+		return p9.QID{}, p9.AttrMask{}, p9.Attr{}, err
+	}
+	id.Qid.Type = id.meta.Mode.QIDType()
+
+	metaClone := id.meta
+	metaClone.Filter(req)
+
+	return id.Qid, req, metaClone, nil
+}
+
+func (id *IPFS) Walk(names []string) ([]p9.QID, p9.File, error) {
+	id.Logger.Debugf("Walk names %v", names)
+	id.Logger.Debugf("Walk myself: %s:%v", id.Path, id.Qid)
+
+	if shouldClone(names) {
+		id.Logger.Debugf("Walk cloned")
+		return []p9.QID{id.Qid}, id, nil
+	}
+
+	qids := make([]p9.QID, 0, len(names))
+
+	walkedNode := &IPFS{} // [walk(5)] id represents fid, walkedNode represents newfid
+	*walkedNode = *id
+
+	var err error
+	for _, name := range names {
+		//TODO: timerctx; don't want to hang forever
+		if walkedNode.Path, err = id.core.ResolvePath(id.Ctx, corepath.Join(walkedNode.Path, name)); err != nil {
+			return qids, nil, err
+		}
+
+		ipldNode, err := id.core.Dag().Get(id.Ctx, walkedNode.Path.Cid())
+		if err != nil {
+			return qids, nil, err
+		}
+
+		//TODO: this is too opague; we want core path => qid, dirent isn't necessary
+		dirEnt := &p9.Dirent{}
+		if err = ipldStat(dirEnt, ipldNode); err != nil {
+			return qids, nil, err
+		}
+
+		walkedNode.Qid = dirEnt.QID
+		//
+		qids = append(qids, walkedNode.Qid)
+	}
+
+	id.Logger.Debugf("Walk ret %v, %v", qids, walkedNode)
+	return qids, walkedNode, err
+}
+
+func (id *IPFS) Open(mode p9.OpenFlags) (p9.QID, uint32, error) {
+	id.Logger.Debugf("Open %q", id.Path)
+
+	// set up  handle amenities
```

Comment:
https://github.com/ipfs/go-ipfs/blob/02f69b81b240f08e546cd6289cd9b87ee11eab09/plugin/plugins/filesystem/nodes/ipfs.go#L100-L102
I meant to change this and forgot. 
Specifically consider `:=` vs `=` 🙈
It should look closer to this
```
	var handleContext context.Context
	handleContext, id.operationsCancel = context.WithCancel(id.Ctx)
```